src/libraries/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -253,11 +253,20 @@ private bool FlushMessageQueue(bool rethrowInNewThread)
 
         // Wait until we hit EOF. This is called from Process.WaitForExit
         // We will lose some information if we don't do this.
-        internal void WaitUtilEOF()
+        internal void WaitUntilEOF()
         {
-            if (_readToBufferTask != null)
+            if (_readToBufferTask is Task task)
             {
-                _readToBufferTask.GetAwaiter().GetResult();
+                task.GetAwaiter().GetResult();
+                _readToBufferTask = null;
+            }
+        }
+
+        internal async ValueTask WaitUntilEOFAsync()
+        {
+            if (_readToBufferTask is Task task)
+            {
+                await task.ConfigureAwait(false);
```

Comment:
Would it work for this to be:
```C#
internal Task WaitUntilEOFAsync()
{
    if (_readToBufferTask is Task task)
    {
        _readToBufferTask = null;
        return task;
    }

    return Task.CompletedTask;
}
```
?

Reply:
The only problem with that approach is that we could be nulling out the field prematurely, so it could result in interesting concurrent access behaviours. It might be worth asking if we should be clearing out the field at all.